shell/browser/notifications/win/windows_toast_notification.cc

```diff
@@ -265,13 +265,136 @@ HRESULT WindowsToastNotification::SetXmlScenarioReminder(IXmlDocument* doc) {
+  RETURN_IF_FAILED(
+      actions_wrapper_node->AppendChild(action_node_tmp.Get(), &action_node));
+
+  // Setup attributes for action
+  ComPtr<IXmlNamedNodeMap> action_attributes;
+  RETURN_IF_FAILED(action_node->get_Attributes(&action_attributes));
+
+  // Create activationType attribute
+  ComPtr<IXmlAttribute> activation_type_attribute;
+  ScopedHString activation_type_str(L"activationType");
+  RETURN_IF_FAILED(
+      doc->CreateAttribute(activation_type_str, &activation_type_attribute));
+
+  ComPtr<IXmlNode> activation_type_attribute_node;
+  RETURN_IF_FAILED(
+      activation_type_attribute.As(&activation_type_attribute_node));
+
+  // Set activationType attribute to system
+  ScopedHString activation_type_value(L"system");
+  if (!activation_type_value.success())
+    return E_FAIL;
+
+  ComPtr<IXmlText> activation_type_text;
+  RETURN_IF_FAILED(
+      doc->CreateTextNode(activation_type_value, &activation_type_text));
+
+  ComPtr<IXmlNode> activation_type_node;
+  RETURN_IF_FAILED(activation_type_text.As(&activation_type_node));
+
+  ComPtr<IXmlNode> activation_type_backup_node;
+  RETURN_IF_FAILED(activation_type_attribute_node->AppendChild(
+      activation_type_node.Get(), &activation_type_backup_node));
+
+  // Add activation type to the action attributes
+  ComPtr<IXmlNode> activation_type_attribute_pnode;
+  RETURN_IF_FAILED(action_attributes.Get()->SetNamedItem(
+      activation_type_attribute_node.Get(), &activation_type_attribute_pnode));
+
+  // Create arguments attribute
+  ComPtr<IXmlAttribute> arguments_attribute;
+  ScopedHString arguments_str(L"arguments");
+  RETURN_IF_FAILED(doc->CreateAttribute(arguments_str, &arguments_attribute));
+
+  ComPtr<IXmlNode> arguments_attribute_node;
+  RETURN_IF_FAILED(arguments_attribute.As(&arguments_attribute_node));
+
+  // Set arguments attribute to dismiss
+  ScopedHString arguments_value(L"dismiss");
+  if (!arguments_value.success())
+    return E_FAIL;
+
+  ComPtr<IXmlText> arguments_text;
+  RETURN_IF_FAILED(doc->CreateTextNode(arguments_value, &arguments_text));
+
+  ComPtr<IXmlNode> arguments_node;
+  RETURN_IF_FAILED(arguments_text.As(&arguments_node));
+
+  ComPtr<IXmlNode> arguments_backup_node;
+  RETURN_IF_FAILED(arguments_attribute_node->AppendChild(
+      arguments_node.Get(), &arguments_backup_node));
+
+  // Add arguments to the action attributes
+  ComPtr<IXmlNode> arguments_attribute_pnode;
+  RETURN_IF_FAILED(action_attributes.Get()->SetNamedItem(
+      arguments_attribute_node.Get(), &arguments_attribute_pnode));
+
+  // Create content attribute
+  ComPtr<IXmlAttribute> content_attribute;
+  ScopedHString content_str(L"content");
+  RETURN_IF_FAILED(doc->CreateAttribute(content_str, &content_attribute));
+
+  ComPtr<IXmlNode> content_attribute_node;
+  RETURN_IF_FAILED(content_attribute.As(&content_attribute_node));
+
+  // Set content attribute to Dismiss
+  ScopedHString content_value(L"Dismiss");
+  if (!content_value.success())
+    return E_FAIL;
```

Comment:
_(nit)_ Not a big deal, but for consistency with the rest of the code:

```suggestion
  RETURN_IF_FAILED(content_value.success());
```

Reply:
@ckerr since this is present in a bunch of other toast code as well - why don't we do it as a follow-up and change all of them at once?

Reply:
@codebytere oh, you're right, it is -- there are several places that use this exact phrasing in extant code. :+1:

Reply:
@ckerr @codebytere I think it may be best to leave this as it is actually. `RETURN_IF_FAIL` will only return if the value is strictly less than 0, but with `success()` returning a boolean, it will never register as failing.